caikit/runtime/server_base.py

```diff
@@ -137,6 +137,43 @@ def interrupt(self, signal_, _stack_frame):
         )
         self.stop()
 
+    def _intercept_interrupt_signal(self) -> None:
+        """Intercept signal handlers to allow the server to stop on interrupt.
+        Calling this on a non-main thread has no effect.
+        This does not override any existing non-default signal handlers,
+        it will call them all in the reverse order they are registered.
+        """
+        self._add_signal_handler(signal.SIGINT, self.interrupt)
+        self._add_signal_handler(signal.SIGTERM, self.interrupt)
+
+    @staticmethod
+    def _add_signal_handler(sig, handler):
+        def nested_interrupt_builder(*handlers):
+            """Build and return an interrupt handler that calls all of *handlers"""
+
+            log.debug("Building interrupt handler: %s", handlers)
+
+            def interrupt(signal_, _stack_frame):
+                for handler in handlers:
+                    # Only call the handler if it is a callable fn that is _not_ a default handler
+                    log.debug("Running interrupt handler: %s", handler)
+                    if (
+                        handler
+                        and callable(handler)
+                        and handler != signal.SIG_DFL
+                        and handler is not signal.default_int_handler
+                    ):
+                        handler(signal_, _stack_frame)
+
+            return interrupt
+
+        try:
+            signal.signal(sig, nested_interrupt_builder(handler, signal.getsignal(sig)))
+        except ValueError:
+            log.warning(
```

Comment:
Do we expect this to indicate a problem? Seems like putting this on `warning` would likely cause confusion if this is a "working as expected" usecase when running either of the servers in a child thread.

Reply:
true, I guess this should be info level